beater/v2_handler.go

```diff
@@ -21,53 +21,65 @@ import (
 	"encoding/json"
 	"net/http"
 
+	"github.com/elastic/beats/libbeat/monitoring"
+
 	"github.com/elastic/apm-server/processor/stream"
 	"github.com/elastic/apm-server/publish"
 
 	"github.com/elastic/beats/libbeat/logp"
 
-	"github.com/pkg/errors"
-
 	"github.com/elastic/apm-server/decoder"
 )
 
-var (
-	errInvalidMetadataFormat = errors.New("invalid metadata format")
-)
+const MethodNotAllowedErrType stream.StreamError = 100
```

Comment:
any reason this isn't at the bottom of the list in `processor/stream/result.go` ?

Reply:
`processor/stream` has no knowledge about the used protocol, it is just reading and processing a stream. The `MethodNotAllowedErrType` reflects that the wrong http method was used.
But I guess the err name is generic enough to move it inside `processor` as it doesn't say anything about the protocol.